src/assign.c

```diff
@@ -1002,6 +1011,11 @@ void writeNA(SEXP v, const int from, const int n)
       for (int i=from; i<=to; ++i) vd[i] = NA_REAL;
     }
   } break;
+  case CPLXSXP: {
+    Rcomplex *vd = COMPLEX(v);
+    Rcomplex NA_CPLX = {NA_REAL, NA_REAL};
```

Comment:
a bit strange why we have to define this ourselves but I see it done like this in a few places in `r-source` (e.g. `subset.c`)

Reply:
Yes I see 3 in src/main/subset.c too.  Those are the only occurence of "NA_C" in the entire source tree of R.  So since it's so rarely used, I guess it just hasn't been looked at.  Which seems reasonable.